cli/src/main/java/com/box/l10n/mojito/cli/command/PushCommand.java

```diff
@@ -146,4 +170,37 @@ public void execute() throws CommandException {
 
         consoleWriter.fg(Ansi.Color.GREEN).newLine().a("Finished").println(2);
     }
+
+    /**
+     * Get fileMatch list contains git changes against git tree HEAD
+     * @param sourceFileMatches
+     * @return
+     * @throws CommandException
+     */
+    private ArrayList<FileMatch> getGitChangeFiles(ArrayList<FileMatch> sourceFileMatches) throws CommandException, GitAPIException {
+        logger.debug("filter out source files which do not contain diff changes.");
+        GitRepository gitRepository = new GitRepository();
+        gitRepository.init(commandDirectories.getSourceDirectoryPath().toString());
+        Git git = new Git(gitRepository.jgitRepository);
+        String gitPath = gitRepository.getDirectory().getPath().replace(".git", "");
+        Set<String> gitDiffFiles = new HashSet<>();
+
+        Pattern pattern = Pattern.compile(gitRegex);
+        for(FileMatch fileMatch: sourceFileMatches) {
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            List<DiffEntry> diff = git.
+                    diff().
+                    setCached(true).
+                    setOutputStream(byteArrayOutputStream).
+                    setPathFilter(PathFilter.create(fileMatch.getPath().toString().substring(gitPath.length()))).
+                    call();
+            if(gitRegex == null || pattern.matcher(byteArrayOutputStream.toString()).find()) {
```

Comment:
will changed to **if ((gitRegex == null && !diff.isEmpty()) || pattern.matcher(byteArrayOutputStream.toString()).find())**

push change together with unit test